### PR TITLE
Player Manager refactoring warnings

### PR DESCRIPTION
**Pull request by m4rtint**

### CODE REVIEW

**Comment by m4rtint** [comment]
[Refactor PlayerManager](https://trello.com/c/PjdD5szT/317-refactor-playermanager)

File: Assets/Script/Managers/PlayerManager.cs [hunk 1]
```diff
@@ -136,15 +137,9 @@ private void SetupPlayerCount()
...
-        try
+        if (m_players.Count < 5 && m_players.Count > 0)
         {
-            Assert.IsTrue(m_players.Count < 5 && m_players.Count > 0);
-        }
-        catch (AssertionException e)
-        {
-            Debug.LogError(e.Message);
-            Debug.LogError("Number of players needs to be between 1-4");
+            throw new Exception(Warnings.NumberOfPlayersWarning);
```
Comment (Gursaanj) [inline Assets/Script/Managers/PlayerManager.cs]: Doesn't this mean that the Exception will be thrown if player count is between 1-4??

Comment (m4rtint) [reply Assets/Script/Managers/PlayerManager.cs]: Yes, changing

File: Assets/Script/Managers/PlayerManager.cs [hunk 2]
```diff
@@ -136,15 +137,9 @@ private void SetupPlayerCount()
 
     private void CheckNumberOfPlayers()
     {
-        Assert.raiseExceptions = true;
-        try
+        if (m_players.Count > 4 && m_players.Count <= 0)
```
Comment (Gursaanj) [inline Assets/Script/Managers/PlayerManager.cs]: Shouldn't this be an OR statement, rather than an AND statement ?

**Pull request merged**